### PR TITLE
Release notes for OpenSearch 1.2.1

### PR DESCRIPTION
**Pull request by peternied**
Signed-off-by: Peter Nied <petern@amazon.com>

### Description
Update the release notes for the new version  

### Issues Resolved
Related to https://github.com/opensearch-project/opensearch-build/issues/1290
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by stockholmux — APPROVED** [review]
LGTM

**Review by gaiksaya — APPROVED** [review]
Thanks!

**Review by elfisher — APPROVED** [review]
lgtm!

File: release-notes/opensearch-release-notes-1.2.1.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# OpenSearch 1.2.1 Release Notes
```
Comment (gaiksaya) [inline release-notes/opensearch-release-notes-1.2.1.md:1]: Is it just opensearch or dashboard too?

Comment (peternied) [reply release-notes/opensearch-release-notes-1.2.1.md:1]: OpenSearch Dashboards was _not_ released for 1.2.1 as it was unaffected

Comment (gaiksaya) [reply release-notes/opensearch-release-notes-1.2.1.md:1]: Looking at previous release notes https://github.com/opensearch-project/opensearch-build/blob/main/release-notes/opensearch-release-notes-1.2.0.md

**Review by bbarani — APPROVED** [review]

**Pull request merged**